### DISCUSSION TITLE
Restrict the number of items to borrow.

### DISCUSSION THREAD

**Discussion by BiblKumar** [textbrowser/biblioteq#107] [Q&A] [answered]
In BiblioteQ there is no option in the members data to restrict to take particular number of books, it is allowing to take any number of books or magazines but not restricting the member to take these many only. If you can add that option in the members database from restricting the member to take 1 to 10 or 20, it would be good to us, so, every time we don't need to check it. Please, can you let me know is that possible to make the changes in the software tool.

**Comment by textbrowser** [comment]
Yes, I know. That responsibility lies on the library. :) Will review the request.

**Reply by BiblKumar** [reply]
Thank you, it would be great if you consider that option and upgrade it.

**Comment by textbrowser** [comment, marked as answer]
Will add the information to Database Enumerations as per-patron limits require much work.

**Reply by BiblKumar** [reply]
Thank you for the reply, It should be linked to the member. If it is not able to restrict to the number of books, at-least it can show the number of books or magazines plan on display when we search for particular member, for this you don't need to do much work I think. In this scenario we simple put the number and save it as well as we can change it whenever a member want decrease or increase number of books or magazines.

**Reply by BiblKumar** [reply]
Simply you can add this display option below the MemberID option in the create membership page.

**Comment by textbrowser** [comment]
All members must have the same limits. Yes? The limits will be defined in the Enumerators panel.

**Reply by BiblKumar** [reply]
No, different members takes different plans...like Member'A' want 1 book at a time and Member 'B' want 10 books at a time and Member 'C' want 3 books at a time.... for everybody is not the same limit...some members don't want lot of books. So, member 'A' is allowed take 1 book after he/she returns it should allow the take another book...they can do number times but he/she can allowed to take it 1 book till the expiry date, in between if the member 'A' want to take additional 1 or 2 books then the plan can altered in the member creation page only, system should not allow to take it directly until the plan got changed. 
   So, if it is difficult to change the software programming according to above process atleast you can add a display option for number of books or magazines plan  in the member creation page.

**Comment by textbrowser** [comment]
The patron does not create the policy of the library. The number of items any given patron can borrow is limited by the policies of a specific library. I'm describing the limits of the library and not the preferences of a patron. If patron A wishes to borrow 5 items and patron B wishes to borrow 10, that's fine. If the limit is set to 15, then both are allowed to borrow their desired items. Now, if patron C wishes to borrow 20, they can only borrow 15 of those 20 per library policy.

**Reply by BiblKumar** [reply]
Thank you, I understand....this works if the library have lot of books ...but if u have like 8k books..when members issued most of the books.. u can just see the empty shelves or else u need to wait till they return it or u need more funds to add more books.
So, I am just asking for display option for the number of books or magazines in the members page, if it is possible.

**Comment by textbrowser** [comment]
Members Browser contains columns describing the number of items reserved. So, the information already exists. Are you requesting total counts? I'm still confused. Are you interested in statistics about a library? Like, the number of physical books? That's a lot more work and there is a reminder for a statistics panel.

Currently, the Members Browser contains the number of items a patron has reserved. For instance, they may have 1 book checked out and 1 magazine.

As a start, I'll add the total number of items reserved per patron. Then, I'll modify the Enumerators browse for limits: Maximum Books, etc. Statistics are more complicated because of copies and queries and new browsers. :)

**Reply by BiblKumar** [reply]
I understand what u said...You are saying that number of books will be displayed when u click on list of members reserved items, so, there is no need to add an option books plan instead you are saying that you are going to add an option in the Enumerators which displays the maximum book restriction or count, but it will apply same for every member. I am not asking for that, I am talking about the plan...let me explain in detail..
this is a private library we will take security deposit of the book. let us assume that
Member'A' came to the library n need 1 book at a time only
Member'B' came to the library n need 3 books at a time only
Member'C' came to the library n need 10 books at a time only
Library has the books which members 'A', 'B' & 'C' want to read, but, if the library provide only one plan 15 books maximum u can take and member has to pay for it, we may loose the some patrons or members. 
Member'A' has is reasons like not Avid reader or doesn't have time to read or not interested more than 1 book or not interested to pay for more books..so, member'A' said i will go for 1 book at a time every time....same goes for the members 'B' & 'C' as well. 
If we ask the member to pay security deposit whenever he/she takes the book or books...it will create lot of financial transactions..
If the library has more members it is not easy for the librarian to remember how much security deposit he paid for how many books..so, he has to check every time which he/she saved about the security deposit/books members data base, it is time consuming process. So, instead of checking every time if there is an option like number of books or magazines in the member's database page that he/she paid for that would be great.
So, different libraries provide different plans. In this software there is no display of number of books that members can take which they had paid for it. 
If you provide the option in the members database at-least for display that the every patron or member can take maximum number of books, it would be great.

**Comment by textbrowser** [comment]
Alright. Each member will be assigned the maximum number of items which they may borrow. The information will reside in the Site tab of Create New Member / Modify Member. What's a reasonable default? The information will also be displayed in the Members Browser. A member will not be allowed to borrow items exceeding the limit. This will be an optional limit as it is site-specific.

The following benefits will not exist in this site-specific solution:
- Per category limits. For example, some libraries allow a maximum of 10 DVDs to be borrowed and a maximum of 50 books. The site-specific limit will group all items.
- Per library policies. Per-patron policies are unfair.
- Modifying the limits for a handful of categories is much easier than modifying limits for hundreds or thousands of patrons.
- From a development perspective, this is a lot of work because it involves individual patrons instead of a library.

**Reply by BiblKumar** [reply]
Thank you for the reply. Forget about the restriction part for each n every member, it is a lot of work to do. Just like it will show the member information display in the member page for the librarian to see , like that it will show maximum items that he/she can borrow from the library after the Last name column, but, it will not restrict taking more items, that will taken care by the librarian. 

![image](https://user-images.githubusercontent.com/89518086/134517710-c3116e47-f0e9-4da8-be0a-307e80bfa24e.png)

**Comment by textbrowser** [comment]
Both approaches are fine. One has more benefits. I prefer implementing a solution which also reminds the librarian of the current state of a borrower's items. Is a default of 25 suitable? I will post an image of the proposed appearance once I start work on it. :)

**Reply by BiblKumar** [reply]
Thank you. Please don't put a number leave it empty, the respective librarian will fill the number, Just like the address and phone number.

**Reply by BiblKumar** [reply]
If it is not possible, then put 1 to 25 numbers.

**Comment by textbrowser** [comment]
A default value simplifies matters. :) Basic numeric fields require numbers. I'll assign a zero to indicate that a default is not set, meaning that a maximum does not exist.

**Reply by BiblKumar** [reply]
Thank you so much.

**Comment by textbrowser** [comment]
Hi. The database schemas have been prepared: added maximum_reserved_books to member table. A new Total Reserved column exists in the Members Browser. A new widget in the member panel. The default value is set to 0, meaning a maximum does not exist. More to come.

**Comment by textbrowser** [comment]
![Screenshot_20211018_095126](https://user-images.githubusercontent.com/10701156/137744704-8659fa7b-a199-4fb4-bcd9-a5c9a88367d1.png)

Limits are not enforced. The librarian is responsible.

**Reply by BiblKumar** [reply]
Thank you

**Comment by textbrowser** [comment]
The next release will also feature an Android build. :)

**Reply by BiblKumar** [reply]
Oh Great. If it is possible please consider the option uploading the members data as well, just like the books CSV upload.

Can you let me know, What is the best RAM size to use the application in windows for smooth performance?

**Comment by textbrowser** [comment]
An import for members?

I don't have specific details regarding computing resources. Are you experiencing problems?

**Reply by BiblKumar** [reply]
Yeah an import CSV file for the members as well...if someone wants to migrate from windows to other operating system, it will be helpful for them. You can implement one more option also for member's reservation history, CSV file export...if any member want to know what are the books they had read, simply export the file and we can give it to them.

No problems, just checking whether need any basic specifications to run the application.

**Comment by textbrowser** [comment]
OK, added the request as a reminder. :) Do not know when I'll implement it. Certainly not for this next release.

**Reply by BiblKumar** [reply]
Thank you.
